barretenberg/cpp/src/barretenberg/vm2/simulation/memory.hpp

```diff
@@ -12,6 +12,8 @@ namespace bb::avm2::simulation {
 struct ValueRefAndTag {
     const MemoryValue& value;
     MemoryTag tag;
+
+    bool operator==(const ValueRefAndTag& other) const { return value == other.value && tag == other.tag; }
```

Comment:
doesn't the default work?

`bool operator==(const ValueRefAndTag& other) const = default;`